release notes/v0.29.0.md

```diff
@@ -0,0 +1,133 @@
+
+You can also configure this behavior with the new `--dns` CLI flag, the `K6_DNS` environment variable, or the `dns` script/JSON option. Only 3 DNS resolution options are exposed in this k6 version, `ttl`, `select`, and `policy`.
+
+Possible `ttl` values are :
+- `0`: no caching at all - each request will trigger a new DNS lookup.
+- `inf`: cache any resolved IPs for the duration of the test run _(the old k6 behavior)_.
+- any time duration like `60s`, `5m30s`, `10m`, `2h`, etc.; if no unit is specified (e.g. `ttl=3000`), k6 assumes milliseconds. _The new default value is `5s`._
+
+Possible `select` values are:
+- `first` - always pick the first resolved IP _(the old k6 behavior)_.
+- `random` - pick a random IP for every new connection _(the new default value)_.
+- `roundRobin` - iterate sequentially over the resolved IPs.
+
+Possible `policy` values are:
+- `preferIPv4`: use IPv4 addresses, if available, otherwise fall back to IPv6.
+- `preferIPv6`: use IPv6 addresses, if available, otherwise fall back to IPv4 _(the new default value)_.
+- `onlyIPv4`: only use IPv4 addresses, ignore any IPv6 ones.
+- `onlyIPv6`: only use IPv6 addresses, ignore any IPv4 ones.
+- `any`: no preference, use all addresses _(the old k6 behavior)_.
+
+Here are some configuration examples:
+
+```bash
+k6 run --dns "ttl=inf,select=first,policy=any" script.js # this is the old k6 behavior
+K6_DNS="select=random,ttl=5m,policy=preferIPv4" k6 cloud script.js # new default behavior
+# syntax for the JSON config file or for the exported script `options`:
+echo '{"dns": {"select": "roundRobin", "ttl": "1h33m7s", "policy": "onlyIPv6"}}' > config.json
+k6 run --config "config.json" script.js
+```
+
+### Support for Go extensions ([#1688](https://github.com/loadimpact/k6/pull/1688))
+
+After some discussions ([#1353](https://github.com/loadimpact/k6/issues/1353)) and exploration of different approaches for Go-based k6 extensions, we've settled on adopting an approach similar to [caddy's](https://caddyserver.com/docs/extending-caddy). In short, [`xk6`](https://github.com/k6io/xk6) (modeled after [`xcaddy`](https://github.com/caddyserver/xcaddy)) is a small stand-alone tool that will allow the building of custom k6 binaries with 3rd party extensions bundled in. The extensions can be simple Git repositories (no central infrastructure needed!) with Go modules. They will be fully compiled, not interpreted, a part of the final custom k6 binary users will be able to build with k6.
+
+xk6 is not yet stable or documented, so any plugin authors will struggle until we stabilize and document everything in the coming weeks. The important part is that the k6 changes that would allow xk6 to work were implemented in [#1688](https://github.com/loadimpact/k6/pull/1688), so k6 v0.29.0 is the first version compatible with xk6!
```

Comment:
In the first paragraph we call them "extensions" and here we call them "plugins". It's even more confusing as in the code they're "modules". We should standardize a single naming convention to avoid confusion, so maybe "plugin" is a better generic term, and internally "modules" makes sense, mainly because they're "JS modules".

Reply:
Fixed by https://github.com/loadimpact/k6/commit/879d2bf6a58ec4a1ba1d9d1b71b0e0019545bc0d?